process_linux.go

```diff
@@ -226,7 +266,14 @@ func (p *initProcess) wait() (*os.ProcessState, error) {
 		return p.cmd.ProcessState, err
 	}
 	// we should kill all processes in cgroup when init is died if we use host PID namespace
-	if p.cmd.SysProcAttr.Cloneflags&syscall.CLONE_NEWPID == 0 {
+	// If the container joins another PID namespace, do not do anything.
+	var joinPidns bool
+	for _, ns := range p.joinNamespaces {
+		if ns.Type == configs.NEWPID {
+			joinPidns = true
+		}
+	}
+	if !joinPidns && p.cmd.SysProcAttr.Cloneflags&syscall.CLONE_NEWPID == 0 {
```

Comment:
done. The error I noticed was because two container in tests share the same cgroups. Changed the tests to use different cgroups for the containers.